--- a/src/components/Header.tsx
+++ b/src/components/Header.tsx
@@ -516,6 +516,8 @@ export default function Header({
                      onClick={() => {
                        setShowAccountPopover(false);
+                        if (!currentUser) onNavigate('login');
+                        else
                        onNavigate('profile');
                      }}
                      className="w-full h-12 px-4 hover:bg-slate-50 text-slate-705 text-slate-700 font-extrabold text-xs flex items-center gap-3.5 transition-colors border-b border-slate-100 last:border-0 rounded-lg cursor-pointer"
@@ -531,6 +533,8 @@ export default function Header({
                      onClick={() => {
                        setShowAccountPopover(false);
+                        if (!currentUser) onNavigate('login');
+                        else
                        onNavigate('my-orders');
                      }}
                      className="w-full h-12 px-4 hover:bg-slate-50 text-slate-700 font-extrabold text-xs flex items-center gap-3.5 transition-colors border-b border-slate-100 last:border-0 rounded-lg cursor-pointer"
@@ -546,6 +550,8 @@ export default function Header({
                      onClick={() => {
                        setShowAccountPopover(false);
+                        if (!currentUser) onNavigate('login');
+                        else
                        onNavigate('wishlist');
                      }}
                      className="w-full h-12 px-4 hover:bg-slate-50 text-slate-700 font-extrabold text-xs flex items-center gap-3.5 transition-colors border-b border-slate-100 last:border-0 rounded-lg cursor-pointer"
@@ -561,6 +567,8 @@ export default function Header({
                      onClick={() => {
                        setShowAccountPopover(false);
+                        if (!currentUser) onNavigate('login');
+                        else
                        onNavigate('notifications');
                      }}
                      className="w-full h-12 px-4 hover:bg-slate-50 text-slate-700 font-extrabold text-xs flex items-center gap-3.5 transition-colors border-b border-slate-100 last:border-0 rounded-lg cursor-pointer"
@@ -576,6 +584,8 @@ export default function Header({
                      onClick={() => {
                        setShowAccountPopover(false);
+                        if (!currentUser) onNavigate('login');
+                        else
                        onNavigate('addresses');
                      }}
                      className="w-full h-12 px-4 hover:bg-slate-50 text-slate-700 font-extrabold text-xs flex items-center gap-3.5 transition-colors border-b border-slate-100 last:border-0 rounded-lg cursor-pointer"
